### ISSUE TITLE
ONBUILD: fails for flags for echo command totally valid SC2039

### ISSUE THREAD

**Issue by marcellodesales** [hadolint/hadolint#281]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

* No errors

### Actual behavior

* `SC2039`

### Steps to reproduce the behavior

**$ hadolint --version Haskell Dockerfile Linter 1.15.0-no-git**

```bash
./runner/openjdk-8/Dockerfile:59 SC2039 In POSIX sh, echo flags are undefined.
```

**Dockerfile** (if relevant)

```Dockerfile
ONBUILD RUN echo "-Djava.security.egd=file:/dev/./urandom" > ${RUNNER_HOOKS_ENV_VAR_DIR}/springboot.opt
```

**Additional environment details (OS, stack version, etc.)**

**Comment by lorenzo** [comment]
This seems to be a bug (or an actual valid warning) in Shellcheck. I'm not knowledgable enough about this topic to form an opinion. Here's the upstream issue https://github.com/koalaman/shellcheck/issues/1011

A workaround I found is adding a space before the dash:

```
echo " -Djava.security.egd=file:/dev/./urandom"
```

**Comment by ribx** [comment]
Also:

```
FROM alpine
RUN /bin/echo -e "#/bin/sh\nls -l" > /usr/bin/ll
```
gives `SC2039 In POSIX sh, echo flags are undefined.` See https://github.com/koalaman/shellcheck/issues/1593

**Comment by zemanlx** [comment]
Using `SHELL` with `bash` for OpenJDK or `ash` for Alpine will force Hadolint's Shellcheck to not assume that you use `sh`.

**Comment by SwuduSusuwu** [comment]
Rather than require `ash` or `bash`, you can replace `echo -e "message"` with `printf "message\n"`, as https://www.shellcheck.net/wiki/SC3037 suggests to
